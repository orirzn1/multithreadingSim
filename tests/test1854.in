L15
S7
O14
N12

T0
I@0x00000000
LOAD $7, $1, 0x2
LOAD $6, $5, $2
STORE $7, $6, $4
SUBI $7, $4, 0x1
HALT $0

T1
I@0x00000000
ADD $1, $1, $1
SUB $0, $4, $3
STORE $0, $7, $6
SUB $1, $4, $6
STORE $4, $1, 0x4
ADD $6, $3, $2
HALT $0

T2
I@0x00000000
SUBI $1, $3, 0x4
SUB $4, $4, $2
SUB $5, $6, $0
SUB $7, $1, $5
SUB $4, $5, $0
ADD $6, $5, $1
SUBI $4, $7, 0x1
HALT $0

T3
I@0x00000000
LOAD $5, $6, $0
SUBI $1, $3, 0x3
ADDI $0, $5, 0x2
LOAD $6, $2, 0x4
SUBI $6, $4, 0x2
SUB $5, $1, $1
SUBI $7, $1, 0x1
ADDI $7, $0, 0x2
LOAD $2, $1, 0x0
HALT $0

T4
I@0x00000000
STORE $2, $0, $1
LOAD $3, $1, $1
ADDI $1, $4, 0x2
ADDI $2, $7, 0x2
ADDI $7, $2, 0x2
STORE $6, $1, $0
SUB $7, $0, $5
ADD $2, $1, $1
LOAD $7, $4, $2
SUBI $1, $0, 0x3
HALT $0

T5
I@0x00000000
ADDI $3, $2, 0x1
LOAD $2, $4, 0x4
ADDI $0, $3, 0x1
ADD $0, $2, $4
SUB $0, $2, $4
STORE $2, $6, $0
ADDI $1, $4, 0x3
HALT $0

T6
I@0x00000000
ADD $7, $7, $6
ADDI $5, $6, 0x4
SUB $0, $7, $2
SUBI $7, $3, 0x2
HALT $0

T7
I@0x00000000
LOAD $1, $2, 0x0
ADDI $0, $6, 0x3
SUBI $2, $6, 0x4
ADD $4, $4, $7
SUBI $2, $1, 0x1
HALT $0

T8
I@0x00000000
SUB $2, $1, $3
SUB $1, $2, $5
ADDI $6, $4, 0x3
LOAD $3, $4, $4
SUB $3, $5, $5
SUB $1, $5, $4
LOAD $4, $0, $7
LOAD $1, $3, 0x0
ADD $5, $5, $5
ADDI $5, $6, 0x3
HALT $0

T9
I@0x00000000
ADDI $7, $4, 0x2
STORE $0, $0, $0
STORE $7, $7, $0
STORE $5, $1, 0x3
STORE $6, $6, $2
ADDI $2, $7, 0x1
LOAD $2, $0, $5
SUB $2, $6, $3
ADD $0, $6, $4
STORE $4, $5, $0
HALT $0

T10
I@0x00000000
ADDI $3, $7, 0x4
STORE $0, $4, 0x0
STORE $6, $6, 0x0
SUBI $5, $0, 0x2
SUB $3, $2, $2
HALT $0

T11
I@0x00000000
ADDI $6, $4, 0x3
STORE $5, $1, 0x2
STORE $1, $0, 0x0
SUBI $1, $0, 0x1
SUB $1, $6, $3
STORE $3, $4, $2
LOAD $5, $2, 0x4
HALT $0

D@0x0
0x4
0x8
0x5
0x7
0x7
0x3
0x5
0x7
0x1
