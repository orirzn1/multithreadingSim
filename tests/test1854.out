
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFB	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0xFFFFFFFC
Register file thread id 3:
	R0 = 0x2	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 4:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x8	R1 = 0x3	R2 = 0x8	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 7:
	R0 = 0x3	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x2	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x6	R6 = 0x3	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Blocked MT CPI for this program 6.906250

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFB	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0xFFFFFFFC
Register file thread id 3:
	R0 = 0x2	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 4:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 7:
	R0 = 0x3	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x6	R6 = 0x3	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Finegrained Multithreading CPI for this program 1.166667

